testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java

```diff
@@ -863,6 +863,19 @@ public void testBucketedUnpartitionedDelete(Engine inserter, Engine deleter)
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testDeleteOverManySplits()
+    {
+        withTemporaryTable("delete_select", true, false, NONE, tableName -> {
+            onTrino().executeQuery(format("CREATE TABLE %s WITH (transactional = true) AS SELECT * FROM tpch.sf10.orders", tableName));
```

Comment:
How long does this test take on CI?
instead of using large data set you probably could reduce split sizes, eg using `max_initial_split_size` session property

can we verify multiple splits are actually created? (`QueryStats` includes this information)

Reply:
This takes about two minutes:
```
Test io.trino.tests.product.hive.TestHiveTransactionalTable::testDeleteOverManySplits took 1.90m
```
I'm going to merge this now and filed https://github.com/trinodb/trino/issues/9476 to follow up